Lib/urllib.py

```diff
@@ -203,7 +203,10 @@ def open(self, fullurl, data=None):
         name = 'open_' + urltype
         self.type = urltype
         name = name.replace('-', '_')
-        if not hasattr(self, name):
+        
+        # bpo-35907: # disallow the file reading with the type not allowed
+        if not hasattr(self, name) or \
+            (self == _urlopener and name == 'open_local_file'):
```

Comment:
Please simplify the fix. There is no need to check for ``_urlopener`` here. Just block all access to ``local_file://`` schema. We don't need special cases for subclasses or special instances.